src/mpi/comm/comm_split_type.c

```diff
@@ -28,6 +28,9 @@ int MPIR_Comm_split_type(MPIR_Comm * user_comm_ptr, int split_type, int key,
     }
 
     if (split_type == MPI_COMM_TYPE_SHARED) {
+        /* NOTE: typical path goes down to device layer rather than here.
+         * Currently ch4 calls MPIR_Comm_split_type_by_node, which checks
+         * the "shmem_topo" infohints */
         mpi_errno = MPIR_Comm_split_type_self(comm_ptr, key, newcomm_ptr);
```

Comment:
The note is a bit confusing to me. Does the typical path mean "MPI_Comm_split_type->MPIR_Comm_split_type_impl"?

I think it might be better to say that the MPIR approach for `TYPE_SHARED` is a fallback used only when the device layer does not handle such a split type.

Reply:
Amended